magicblock-chainlink/src/chainlink/fetch_cloner.rs

```diff
@@ -0,0 +1,2609 @@
+                                            ))
+                                        {
+                                            programs.push((
+                                                pubkey,
+                                                account_shared_data,
+                                                slot,
+                                            ));
+                                        } else {
+                                            warn!(
+                                                "Not cloning native loader program account: {pubkey} (should have been blacklisted)",
+                                            );
+                                        }
+                                    } else {
+                                        plain.push((
+                                            pubkey,
+                                            account_shared_data,
+                                        ));
+                                    }
+                                }
+                                ResolvedAccount::Bank(pubkey) => {
+                                    in_bank.push(pubkey);
+                                }
+                            };
+                        }
+                    }
+                    (not_found, in_bank, plain, owned_by_deleg, programs)
+                },
+            );
+
+        if log::log_enabled!(log::Level::Trace) {
+            let not_found = not_found
+                .iter()
+                .map(|(pubkey, slot)| (pubkey.to_string(), *slot))
+                .collect::<Vec<_>>();
+            let in_bank = in_bank
+                .iter()
+                .map(|(p, _)| p.to_string())
+                .collect::<Vec<_>>();
+            let plain =
+                plain.iter().map(|(p, _)| p.to_string()).collect::<Vec<_>>();
+            let owned_by_deleg = owned_by_deleg
+                .iter()
+                .map(|(pubkey, _, slot)| (pubkey.to_string(), *slot))
+                .collect::<Vec<_>>();
+            let programs = programs
+                .iter()
+                .map(|(p, _, _)| p.to_string())
+                .collect::<Vec<_>>();
+            trace!(
+                "Fetched accounts: \nnot_found:      {not_found:?} \nin_bank:        {in_bank:?} \nplain:          {plain:?} \nowned_by_deleg: {owned_by_deleg:?}\nprograms:       {programs:?}",
+            );
+        }
+
+        let (clone_as_empty, not_found) =
+            if let Some(mark_empty) = mark_empty_if_not_found {
+                not_found
+                    .into_iter()
+                    .partition::<Vec<_>, _>(|(p, _)| mark_empty.contains(p))
+            } else {
+                (vec![], not_found)
+            };
+
+        // For accounts we couldn't find we cannot do anything. We will let code depending
+        // on them to be in the bank fail on its own
+        if !not_found.is_empty() {
+            warn!(
```

Comment:
Moving it to debug is a good idea, since this a very frequent event

Reply:
Addressed in 1a371038